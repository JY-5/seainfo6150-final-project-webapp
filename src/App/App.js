import React, { useEffect, useState } from "react";
import { Switch, Route, Link } from "react-router-dom";
import { isEmpty } from "lodash";
import Home from "../components/Home/Home.jsx";
import Error from "../components/Error/Error.jsx";
import Login from "../components/Login/Login";
import logo from "../assets/images/logo.png";
import Recipe from "../components/Recipe/Recipe";
import AddRecipe from "../components/AddRecipe/AddRecipe";
import Footer from '../components/Footer/Footer';
import styles from './App.module.css';
import SearchResult from '../components/SearchResult/SearchResult';
import { withRouter } from 'react-router';

function App(props) {
  const [categories, setCategories] = useState({});
  const [popularRecipes, setPopularRecipes] = useState({"0": {title: '', id: '', url: ''}}); 
  const[ifLoggedIn, setIfLoggedIn] = useState(false);
  const [searchInput, setSearchInput] = useState(''); 
  const [username, setUsername] = useState('');
  const [allRecipes, setAllRecipes] = useState();

  useEffect(() => {
    const fetchCategories = async () => {
      const response = await fetch("https://run.mocky.io/v3/b7be0f81-1822-4557-8f68-f65c022a2966");
      const responseJson = await response.json();
      setCategories(responseJson["Categories"]);
    };

    if (isEmpty(categories)) {
        fetchCategories();
    }

    const fetchPopularRecipes = async () => {
      const response = await fetch("https://run.mocky.io/v3/f5ee7031-c75e-41ea-9540-d59408af8803");
      const responseJson = await response.json();
      setPopularRecipes(responseJson);
      };
      fetchPopularRecipes();

      const fetchRecipes = async () => {
        const response = await fetch("https://run.mocky.io/v3/9d96b81a-b6e2-4c25-bbcd-63772f60140f");
        const responseJson = await response.json();
        setAllRecipes(responseJson);
    };
    fetchRecipes();
    }, []);

    const changeLogIn = (username) => {
      setIfLoggedIn(!ifLoggedIn);
      setUsername(username);
    }

  const onChangeSearch = (e) => {
    setSearchInput(e.target.value);
  }

  return (
    <div className={styles.app}>     
          <nav className={styles.nav}>
            <ul className={styles.ul}>
              <li className={styles.logoPanel}>
                <Link to="/" className={styles.logoLink}>
                  <img className={styles.logo} src={logo} alt="treasured recipes"/>
                </Link>
                <Link to="/" className={styles.logoName}>
                  <span className={styles.name}>Recipes</span>
                </Link>
              </li>

              <li className={styles.searchPanel}>
                <input className={styles.search} type="text" name="search" onChange={onChangeSearch} value={searchInput}/>
                <Link to={`/recipes/${searchInput}`} className={styles.searchLink} >
                  Search
                </Link>
              </li>
              <li className={styles.addLi}>
                <Link to="/newRecipe" className={styles.addNew}>
                  Add a new recipe
                </Link>
              </li>  
              <li className={styles.logLi}>      
                {ifLoggedIn ?
                  <button onClick={changeLogIn} className={styles.logout}>Log out</button> :
                  <Link to="/login" className={styles.login}>Log in</Link>
                }
              </li>
            </ul>
          </nav>
      <div className={styles.right}>
      <Switch>
        <Route path="/login" exact render={() => <Login changeLogIn={changeLogIn} />}/>
        <Route path="/recipes/:name" render={({ match }) => <SearchResult searchWord={searchInput} header={`Recipes result of ${match.params.name}`}/>} />
        <Route path="/recipes/" render={() => <SearchResult searchWord={searchInput} header='All recipes: '/>} />
        <Route path="/newRecipe" exact render={() => (
          ifLoggedIn ?
          <AddRecipe username={username} ifLoggedIn={ifLoggedIn}/> : props.history.push('/login'))}/> 
        <Route 
          path="/recipe/:recipeId"
          exact
          render={({ match }) => (
            <Recipe
              recipeId={match.params.recipeId} recipe={popularRecipes[match.params.recipeId]}
            />
          )}
        />
        <Route 
          path="/categories/:categoryId/:recipeId"
          exact
          render={({ match }) => (
            <Recipe
              recipeId={match.params.recipeId} recipe={allRecipes[match.params.categoryId][match.params.recipeId]} 
            />
          )}
        />
        <Route path="/" render={() => <Home categories={categories} popularRecipes={Object.values(popularRecipes)} allRecipes={allRecipes}/>}/>
        <Route path="/404page" component={Error} />
      </Switch>
      </div>
      <div className={styles.footer}>
        <Footer/>
      </div>
    </div>
  );
}

export default withRouter(App);
