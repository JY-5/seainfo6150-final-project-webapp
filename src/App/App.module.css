body {
    margin-top: 13rem;
}

.nav {
    /* border-bottom: 1px solid rgb(156, 155, 155); */
    box-shadow: 0 4px 6px -2px rgba(221,221,221,0.64);
    /* margin-bottom: 1rem; */
    /* justify-content: space-between; */
    position: fixed; 
    width: 100%;
    top: 0;
    background-color: white;
}

.logoPanel {
    display: flex;
}

.logoLink {
    align-self: center;
}

.logoName {
    text-decoration: none;
}

.name {
    font-family: 'Lobster', cursive;
    color: #6eb339;
    font-size: 3rem;
}

.ul {
    /* 
            padding: 0.4rem 2rem;

    display: flex;
    flex-flow: row wrap;
    justify-content: space-between;
 */
        display: flex;
        flex-flow: column wrap;
        justify-content: space-between;
        list-style: none;
        font-family: 'Prompt', sans-serif;
        margin-top: 0;
        margin-bottom: 0;
        padding: 0.4rem 0;
        height: 13rem;
        /* width: 100%; */

}

.ul li {
    /* align-self: center; */
}

.logo {
    width: 2.7rem;
    height: 2.7rem;
    margin-right: 0.2rem;
}

.addNew {
    text-decoration: none;
    font-size: 1.1rem;
    color: white;
    background-color: #6eb339;
    padding: 0.2rem;
    border-radius: 4px;
}

.login {
    text-decoration: none;
    color: #413e3e;
    font-size: 1.1rem;
}

.home {
    display: grid;
    grid-template-rows: 1fr;
    /* grid-template-columns: 13rem 1fr; */
    grid-template-columns: 1fr 9fr;

    grid-gap: 2.5rem;
    /* display: flex;
    flex-flow: row wrap; */
}

.categories {
    /* grid-row: 1 / 4;
    grid-column: 1 / 2; */
    margin-left: 2rem;

}
.searchPanel {
    display: flex;
    flex-flow: column wrap;
}
.searchLink {

} 

.search {
    height: 1.4rem;
    width: 15rem;
    margin-bottom: 0.2rem;
}

.searchLink {
    color: white;
    text-decoration: none;
    /* margin-left: 1rem; */
    background-color: #6eb339;
    border-radius: 4px;
    padding: 0.2rem;
    font-size: 1.1rem;
    flex: 1;
}

.footer {
    text-align: center;
    height: 2rem;
    background-color: #fbfbfb;
    margin-top: 10rem;
}

@media screen and (min-width: 220px) {

    body {
        /* margin-top: 12rem; */
    }
    .ul {
    }
    /* */
}
  
@media screen and (min-width: 1000px) {
    .recommendation {
        grid-template-columns: repeat(3, 1fr);
    }
}

@media screen and (min-width: 1200px) {
    .recommendation {
        grid-template-columns: repeat(4, 1fr);
    }
}