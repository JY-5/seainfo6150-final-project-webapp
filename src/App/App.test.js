import React from "react";
import App from './App';
import { render } from "@testing-library/react";
import {BrowserRouter as Router} from "react-router-dom";

describe("Add App tests", () => {
    it("renders correctly", () => {
      const { container } = render(<Router><App/></Router>);
      expect(container).toMatchSnapshot();
    });
});