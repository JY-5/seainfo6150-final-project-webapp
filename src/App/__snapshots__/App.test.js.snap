// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add App tests renders correctly 1`] = `
<div>
  <div
    class="app"
  >
    <header>
      <nav
        class="nav"
      >
        <ul
          class="ul"
        >
          <li
            class="logoPanel"
          >
            <a
              class="logoLink"
              href="/"
            >
              <img
                alt="treasured recipes"
                class="logo"
                src="logo.png"
              />
            </a>
            <a
              class="logoLink"
              href="/"
            >
              <span
                class="name"
              >
                Recipes
              </span>
            </a>
          </li>
          <li>
            <input
              class="search"
              name="search"
              type="text"
              value=""
            />
            <a
              class="searchLink"
              href="/recipes/"
            >
              Search
            </a>
          </li>
          <li>
            <a
              class="addNew"
              href="/newRecipe"
            >
              Add a new recipe
            </a>
          </li>
          <li>
            <a
              class="login"
              href="/login"
            >
              Log in
            </a>
          </li>
        </ul>
      </nav>
    </header>
    <div
      class="right"
    >
      <div
        class="home"
      >
        <div
          class="left"
        >
          <nav>
            <h2
              class="header"
            >
              Recipe Categories
            </h2>
            <ul
              class="list"
            />
          </nav>
        </div>
        <div
          class="right"
        >
          <div>
            <div>
              <header
                class="header"
              >
                Personalized Recommendation
              </header>
              <ul
                class="recommendation"
              >
                <li
                  class="item"
                >
                  <a
                    class="title"
                    href="/recipe/"
                  >
                    <img
                      alt=""
                      class="image"
                      src=""
                    />
                  </a>
                  <a
                    class="link"
                    href="/recipe/"
                  />
                </li>
              </ul>
            </div>
            <div>
              <header
                class="header"
              >
                Popular Recipes
              </header>
              <ul
                class="recommendation"
              >
                <li
                  class="item"
                >
                  <a
                    class="title"
                    href="/recipe/"
                  >
                    <img
                      alt=""
                      class="image"
                      src=""
                    />
                  </a>
                  <a
                    class="link"
                    href="/recipe/"
                  />
                </li>
              </ul>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="footer"
    >
      <div
        class="container"
      >
        Contact Us: (408) 123-4567
      </div>
    </div>
  </div>
</div>
`;
